src/core/data/field/field.hpp

```diff
@@ -71,6 +72,25 @@ class Field : public NdArrayView<dim, Data_t>
 
     bool isUsable() const { return Super::data() != nullptr; }
     bool isSettable() const { return !isUsable(); }
+    
+    template<typename GridLayout>
+    void initialize(GridLayout const& layout,
```

Comment:
this is somewhat less flexible than a free floating function, or a templated static function

now it *only* works with instances of Fields

altnernatively it could be like 

```
class FieldUserFunctionInitializer {
  public:

        template<typename Field, typename GridLayout>
        void static initializeComponent_(Field& field, GridLayout const& layout,
                                  initializer::InitFunction<dimension> const& init)
        {
            ...
        }
};
```

Reply:
have a look at the file structure on [my PR ](https://github.com/PHAREHUB/PHARE/pull/888/files) for inspiration to make a new file for the above class

Reply:
Thank you, I will !